lib/checkstl.cpp

```diff
@@ -1637,23 +1637,47 @@ void CheckStl::missingComparison()
             }
 
             const Token *incrementToken = nullptr;
+            bool bComparedInAdvance = false;
 
             // Parse loop..
-            for (const Token *tok3 = scope.bodyStart; tok3 != scope.bodyEnd; tok3 = tok3->next()) {
-                if (Token::Match(tok3, "%varid% ++", iteratorId))
-                    incrementToken = tok3;
-                else if (Token::Match(tok3->previous(), "++ %varid% !!.", iteratorId))
-                    incrementToken = tok3;
-                else if (Token::Match(tok3, "%varid% !=|==", iteratorId))
-                    incrementToken = nullptr;
+            for (const Token *tok3 = scope.bodyStart; tok3 != scope.bodyEnd; tok3 = tok3->next()) 
+            {
+                if (tok3->varId() == iteratorId)
+                {
+                    if (Token::Match(tok3, "%varid% = %name% . insert ( ++| %varid% ++| ,", iteratorId)) 
+                    {
+                        // skip insertion..
+                        tok3 = tok3->linkAt(6);
+                        if (!tok3)
+                            break;
```

Comment:
I don't understand this logic. The 6th token is always going to be '++' if expression to Match() function matches.  It doesn't look like we will ever find link there. But I may be wrong.

Reply:
I agree this is strange. Probably we wanted to find the ")".

Reply:
I think one can use syntax tree here too. I will do that in separate PR to keep things simple.